devicetypes/zooz/zooz-zse11-q-sensor.src/zooz-zse11-q-sensor.groovy

```diff
@@ -0,0 +1,439 @@
+@Field static int tempSensorType = 1
+@Field static int lightSensorType = 3
+@Field static int humiditySensorType = 5
+@Field static int motionSensorType = 12
+
+metadata {
+	definition (
+		name: "Zooz ZSE11 Q Sensor",
+		namespace: "Zooz",
+		author: "Kevin LaFramboise (@krlaframboise)",
+		ocfDeviceType: "x.com.st.d.sensor.motion",
+		mnmn: "SmartThingsCommunity",
+		vid: "42067896-6424-3a34-b753-b87d8c92262f"
+	) {
+		capability "Sensor"
+		capability "Motion Sensor"
+		capability "Tamper Alert"
+		capability "Temperature Measurement"
+		capability "Illuminance Measurement"
+		capability "Relative Humidity Measurement"
+		capability "Battery"
+		capability "Refresh"
+		capability "Health Check"
+		capability "Power Source"
+
+		// zw:Ss2 type:0701 mfr:027A prod:0200 model:0006 ver:1.09 zwv:6.04 lib:03 cc:5E,6C,55,98,9F sec:86,72,71,59,85,80,84,73,30,31,70,5A,7A
+		fingerprint mfr:"027A", prod:"0200", model:"0006", deviceJoinName: "Zooz Multipurpose Sensor" // Zooz ZSE11 Q Sensor (EU)
+		// zw:Ss2 type:0701 mfr:027A prod:0201 model:0006 ver:1.09 zwv:6.04 lib:03 cc:5E,6C,55,98,9F sec:86,72,71,59,85,80,84,73,30,31,70,5A,7A
+		fingerprint mfr:"027A", prod:"0201", model:"0006", deviceJoinName: "Zooz Multipurpose Sensor" // Zooz ZSE11 Q Sensor (US)
+		// zw:Ss2 type:0701 mfr:027A prod:0202 model:0006 ver:1.09 zwv:6.04 lib:03 cc:5E,6C,55,98,9F sec:86,72,71,59,85,80,84,73,30,31,70,5A,7A
+		fingerprint mfr:"027A", prod:"0202", model:"0006", deviceJoinName: "Zooz Multipurpose Sensor" // Zooz ZSE11 Q Sensor (AU)
+	}
+
+	preferences {
+		configParams.each { param ->
+			if (param.options) {
+				input "configParam${param.num}", "enum",
+					title: "${param.name}:",
+					required: false,
+					displayDuringSetup: false,
+					defaultValue: param.defaultVal,
+					options: param.options
+			} else if (param.range) {
+				input "configParam${param.num}", "number",
+					title: "${param.name}:",
+					required: false,
+					displayDuringSetup: false,
+					defaultValue: param.defaultVal,
+					range: param.range
+			}
+		}
+
+		input "tempOffset", "decimal",
+			title: "Temperature Offset:",
+			required: false,
+			defaultValue: 0,
+			range: "-50..50"
+
+		input "humidityOffset", "number",
+			title: "Humidity Offset:",
+			required: false,
+			defaultValue: 0,
+			range: "-50..50"
+
+		input "lightOffset", "number",
+			title: "Light Offset:",
+			required: false,
+			defaultValue: 0,
+			range: "-20000..20000"
+	}
+}
+
+def installed() {
+	log.debug "installed()..."
+	state.firstConfig = true
+	initialize()
+}
+
+def updated() {
+	log.debug "updated()..."
+
+	initialize()
+
+	if (!state.firstConfig) {
+		if (device.currentValue("powerSource") == "battery") {
+			logForceWakeupMessage("Configuration changes will be sent to the device the next time it wakes up.")
+		} else {
+			sendHubCommand(getConfigCmds())
+		}
+	} else {
+		sendHubCommand(getRefreshCmds())
+		state.firstConfig = false
+	}
+}
+
+void initialize() {
```

Comment:
same note as the other DTH

if you put this in `installed` it's guaranteed to only be called once, so you can leave out these if statements.

Reply:
Same reason as the other DTH...

The onboarding process for DTHs has been horrible ever since drivers launched and I have no clue why, but preventing the excess commands seems to make a difference...